### PR TITLE
:sparkles: CLM 254 Update Assessment Preview

### PR DESCRIPTION
**Pull request by Beulah-Matt**
# Description
   I have added the preview functionality by: 

-  Reading from an assessment's configurations for both retrial and question view mode.
-  Initialising form controls only on preview click to avoid empty form values.
-  Rendering an assessment as the learner would see it on Goomza MicroApp.
- Displaying the results after clicking submit. 
- Since this needs no data saving, I have implemented this changes purely on the UI.

Fixes [CLM-254]

## Type of change
- [x] New feature (non-breaking change which adds functionality)

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings

[CLM-254]: https://elewake.atlassian.net/browse/CLM-254?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: libs/features/convs-mgr/conversations/assessments/src/lib/components/assessment-config/assessment-config.component.ts [hunk 1]
```diff
@@ -20,52 +20,130 @@ export class AssessmentConfigComponent implements OnInit, OnDestroy
...
+      if (value.criteria === this.moveOnComplete) {
+        this.clearMoveOnPassControls();
+      } else if (value.criteria === this.moveOnPass) {
+        this.clearMoveOnCompleteControls();
+      }
+    });
 
-      if (!value) {
-        this.clearControls();
+    this._sbS.sink = this.retryTypeControl.valueChanges.subscribe(value => {
```
Comment (ReaganCn) [inline libs/features/convs-mgr/conversations/assessments/src/lib/components/assessment-config/assessment-config.component.ts:52]: Thought you had already replaced these two valuechanges with the toggle logic

File: libs/features/convs-mgr/conversations/assessments/src/lib/pages/assessment-preview-page/assessment-preview-page.component.ts [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+      this.currentProgress.outcome = AssessmentStatusTypes.Failed;
+    }
+
+    if (this.allowedAttempts > 0 && this.retryType === RetryType.OnScore && !passed) {
+      this.resultsMode.failedAndHasRetries = true;
+    } else if (this.allowedAttempts > 0 && this.retryType === RetryType.OnScore && passed) {
+      console.log(this.allowedAttempts)
+      this.resultsMode.passedAndHasRetries = true;
+    } else if (this.allowedAttempts > 0 && this.retryType === RetryType.onCount) {
+      console.log(this.allowedAttempts)
```
Comment (ReaganCn) [inline libs/features/convs-mgr/conversations/assessments/src/lib/pages/assessment-preview-page/assessment-preview-page.component.ts:182]: Remove these console.logs

**Review by ReaganCn — CHANGES_REQUESTED** [review]

File: libs/features/convs-mgr/conversations/assessments/src/lib/pages/assessment-preview-page/assessment-preview-page.component.ts (hunk 2 above)
Comment (Beulah-Matt) [reply libs/features/convs-mgr/conversations/assessments/src/lib/pages/assessment-preview-page/assessment-preview-page.component.ts:182]: Done

File: libs/features/convs-mgr/conversations/assessments/src/lib/components/assessment-config/assessment-config.component.ts (hunk 1 above)
Comment (Beulah-Matt) [reply libs/features/convs-mgr/conversations/assessments/src/lib/components/assessment-config/assessment-config.component.ts:52]: Seems like I had lost the changes. Corrected them now.

**Review by ReaganCn — APPROVED** [review]

**Pull request merged**